Db/cert ssm

derekwbrown opened the pull request:
### What does this PR do?

Changes build process to use certs from SSM. 

### Motivation

Expiring certificate
better security on certificate

### Additional Notes

Anything else we should know when reviewing?

derekwbrown committed:
Build windows by pulling certificate from SSM rather than install on
device.

Fix environment setting/getting

change parameter names to test values in build-stable
Fix unsetting of environment variable

Change check from `SIGN_WINDOWS` to SIGN_PFX

Fix return value for the signing cert (return the filename)

Fix formatting of string

Change ssm keys to "real" keys.
Add code for handling the fact that the b64 encoded key is split across
two values

Remove extra debugging

derekwbrown committed:
Update tests to check for right certificate hash

ogaca-dd commented on tasks/agent.py:
```diff
@@ -319,12 +321,33 @@ def omnibus_build(ctx, puppy=False, log_level="info", base_dir=None, gem_path=No
+            # get certificate and password from ssm
+            pfxfile = get_signing_cert(ctx)
+            pfxpass = get_pfx_pass(ctx)
+            print("Retrieved file {} pass {}".format(pfxfile, pfxpass))
```
Do you want to display `pfxpass`?

hush-hush replied:
Good question !

derekwbrown replied:
oof.  Good catch.  you're right.

ogaca-dd commented on tasks/agent.py:
```diff
@@ -319,12 +321,33 @@ def omnibus_build(ctx, puppy=False, log_level="info", base_dir=None, gem_path=No
+            raise
+
+        if pfxfile:
+            os.remove(pfxfile)
```
I do not know how sensitive is this file (It might because of `mkstemp`) but if you really want to remove it, you can start the `try` at line 325
```python
try:
   if sys.platform == 'win32' and os.environ.get('SIGN_WINDOWS'): // Line 325
   ....
  ctx.run(cmd.format(**args), env=env) // line 342
finally:
   if pfxfile:
     os.remove(pfxfile)
```

ogaca-dd reviewed (commented):
Please check at least my first comment.

derekwbrown committed:
Don't log password
Move try block as suggested so that we're sure to delete the pfx when
we're done, success or failure
